backend/infrahub/core/diff/merger/serializer.py

```diff
@@ -104,7 +104,8 @@ async def serialize_diff(
                 attribute_diff, attribute_property_diff = self._serialize_attribute(
                     attribute_diff=attr_diff, node_uuid=node.uuid, node_kind=node.kind
                 )
-                attribute_diffs.append(attribute_diff)
+                if attribute_diff:
+                    attribute_diffs.append(attribute_diff)
```

Comment:
Not related to this PR but in this context I think it's a bit confusing to have both `attr_diff` and `attribute_diff` unless there's something I'm missing I think we should have an issue to choose other names in the future for these two variables or at least one of them.

Reply:
agreed. that is a bad name. will update it in my next PR